synapse/rest/client/room.py

```diff
@@ -587,14 +621,25 @@ class RoomMessageListRestServlet(RestServlet):
     def __init__(self, hs: "HomeServer"):
         super().__init__()
         self._hs = hs
+        self.clock = hs.get_clock()
         self.pagination_handler = hs.get_pagination_handler()
         self.auth = hs.get_auth()
         self.store = hs.get_datastores().main
 
-    @messsages_response_timer.time()
     async def on_GET(
         self, request: SynapseRequest, room_id: str
     ) -> Tuple[int, JsonDict]:
+        processing_start_time = self.clock.time_msec()
+        # Fire and forget and hope that we get a result by the end.
+        #
+        # `get_number_joined_users_in_room(...)` returns an int so we can type
+        # this as a `Deferred[int]` instead of an optional.
+        room_member_count_deferred: defer.Deferred[int] = run_as_background_process(
```

Comment:
Is this type/cast thing okay?

Reply:
yes, but not sure what 'instead of an optional' is meant to mean here; why would we expect it to be an optional in the first place?

I think this should be `run_in_background`. I think that both `run_as_background_process` and `run_in_background`  are type-safe so I don't think we need to make specific remarks about the type here.

Reply:
> yes, but not sure what 'instead of an optional' is meant to mean here; why would we expect it to be an optional in the first place?

It was because `run_as_background_process` typed the return that way, https://github.com/matrix-org/synapse/blob/22ea51faf9a29cd9b96189d1fac95eaf2cfcf4ec/synapse/metrics/background_process_metrics.py#L191-L197

But I've switched to `run_in_background` now 👍